### PR TITLE
docs: Add explanation on the Hello-world tutorial

### PR DESCRIPTION
**Pull request by SamirPS**
This PR add little explanation about the rockcraft.yaml file on the hello world tutorial

Signed_Off_Based: Samir Akarioh <samir.akarioh@canonical.com>

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

**Comment by SamirPS** [comment]
https://canonical-rockcraft--323.com.readthedocs.build/en/323/

File: docs/tutorials/hello-world.rst [hunk 1]
```diff
@@ -30,6 +30,12 @@ save it as ``rockcraft.yaml``:
 .. literalinclude:: code/hello-world/rockcraft.yaml
     :language: yaml
 
+This file permit to build a ROCK with only the ``hello`` package slice inside,
+using :ref:`chisel_explanation`. For more information
+about the ``parts`` section, check :ref:`ref_parts`.
+The other fields of the file are `metadata
+<https://canonical-rockcraft.readthedocs-hosted.com/en/latest/reference/
```
Comment (dboddie) [inline docs/tutorials/hello-world.rst:37]: Would it be possible to use
> :doc:`/reference/rockcraft.yaml`
instead of a link to the site?

Comment (SamirPS) [reply docs/tutorials/hello-world.rst]: Yeah, it's possible and done

File: docs/tutorials/hello-world.rst [hunk 2]
```diff
@@ -30,6 +30,11 @@ save it as ``rockcraft.yaml``:
 .. literalinclude:: code/hello-world/rockcraft.yaml
     :language: yaml
 
+This file permit to build a ROCK with only the ``hello`` package slice inside,
```
Comment (cjdcordeiro) [inline docs/tutorials/hello-world.rst:33]: fix spelling

File: docs/tutorials/hello-world.rst [hunk 3]
```diff
@@ -30,6 +30,11 @@ save it as ``rockcraft.yaml``:
...
+using :ref:`chisel_explanation`. For more information about the ``parts``
+section, check :ref:`ref_parts`. The other fields of the file are metadata
+which describes the ROCK; for more information about the available check
```
Comment (cjdcordeiro) [inline docs/tutorials/hello-world.rst:36]: this sentence doesn't read well. seems to be missing something..."available..." feature? fields?

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
TY

It would also be nice to add comments to the yaml file, highlighting those divisions (i.e. the metadata from the parts)

File: docs/tutorials/hello-world.rst [hunk 4]
```diff
@@ -30,6 +30,11 @@ save it as ``rockcraft.yaml``:
 .. literalinclude:: code/hello-world/rockcraft.yaml
     :language: yaml
 
+This file permits to build a ROCK with only the ``hello`` package slice inside,
+using :ref:`chisel_explanation`. For more information about the ``parts``
+section, check :ref:`ref_parts`. The other fields of the file are metadata
+which describes the ROCK; for more information about the available fields check
+:doc:`/reference/rockcraft.yaml`
```
Comment (cjdcordeiro) [inline docs/tutorials/hello-world.rst:37]:
```suggestion
This file instructs Rockcraft to build a ROCK that **only** has the ``hello``
binaries package slice (and its dependencies) inside, using
:ref:`chisel_explanation`. For more information about the ``parts``
section, check :ref:`ref_parts`. The remaining YAML fields correspond
to metadata that help define and described the ROCK. For more
information about all available fields check :doc:`/reference/rockcraft.yaml`.
```

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]

**Review by dboddie — APPROVED** [review]

**Review by cjdcordeiro — APPROVED** [review]

**Review by lengau — APPROVED** [review]

**Pull request merged**